pipeline/config.go

```diff
@@ -53,12 +59,27 @@ func (c Config) BuildPipeline(bc operator.BuildContext, defaultOperator operator
 	return NewDirectedPipeline(operators)
 }
 
-func getBuildContextWithDefaultOutput(configs []operator.Config, i int, bc operator.BuildContext) operator.BuildContext {
-	if i+1 >= len(configs) {
-		return bc
+func SetOutputIDs(operators []operator.Operator, pluginMap map[string]string) error {
+	for i, op := range operators {
+		if len(op.GetOutputIDs()) == 0 && i+1 < len(operators) {
+			err := op.SetOutputIDs([]string{operators[i+1].ID()})
+			if err != nil {
+				return err
+			}
+			continue
+		}
+		allOutputs := []string{}
+		for _, id := range op.GetOutputIDs() {
+			if pluginMap[id] != "" {
+				allOutputs = append(allOutputs, pluginMap[id])
+				continue
+			}
+			allOutputs = append(allOutputs, id)
```

Comment:
I know I'm always asking you to flatten your code, but in this case it's probably easier not to escape early.

Also, you can capture the value from the map and check if the key exists at the same time with the `val, ok := map[key]` idiom.

```suggestion
			if pid, ok := pluginMap[id]; ok {
				id = pid
			}
			allOutputs = append(allOutputs, id)
```

Reply:
Changed the logic around to the better logic. I'll keep this in the mind in the future thanks!